File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0){
            return false;
        }
        long int temp,sum=0,r;
            temp=x;
        while(x!=0){
            r=x%10;
            sum=(sum*10)+r;
            x=x/10;
        }
        if(temp==sum)
            return true;
        
        else
            return false;
    }
};

File: 9-palindrome-number/palindrome-number.java
import java.util.Scanner;

class Solution {
    public boolean isPalindrome(int x) {
    Scanner sc=new Scanner(System.in);
    int rev=0;
    int temp=x;
    while(x>0){
        int ld=x%10;
        rev=(rev*10)+ld;
        x=x/10;
    }
    if(rev==temp) return true;
    else 
    return false;
    
          
    }
}